src/DependencyInjection/Greg0ireEnumExtension.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Greg0ire\EnumBundle\DependencyInjection;
+
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * This is the class that loads and manages your bundle configuration
+ *
+ * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html}
+ */
+class Greg0ireEnumExtension extends Extension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $container->setParameter('greg0ire_enum.chain', $config['chain']);
```

Comment:
Why not inject the callable directly in the `BaseEnum` class ? This class is not built with the DIC, so there is no point in creating a container parameter, is there ?